### PR TITLE
Add langchain4j-tablestore Integration:  TablestoreEmbeddingStore/TablestoreChatMemoryStore

### PR DESCRIPTION
**Pull request by xjtushilei**
## Change
Add langchain4j-tablestore Integration:  TablestoreEmbeddingStore / TablestoreChatMemoryStore
Documentation: https://docs.langchain4j.dev/integrations/embedding-stores/tablestore

## General checklist
- [x] There are no breaking changes
- [x] I have added unit and integration tests for my change
- [x] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [x] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green

- [x] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)

## Checklist for adding new embedding store integration

- [x] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [x] I have added my new module in the [BOM](https://github.com/langchain4j/langchain4j/blob/main/langchain4j-bom/pom.xml)

### CODE REVIEW

**Comment by xjtushilei** [comment]
Hello, is there anyone who can help me with the PR? @langchain4j

**Comment by langchain4j** [comment]
Hi @xjtushilei, thank you for your PR! I will try to review it this week

File: docs/docs/integrations/embedding-stores/tablestore.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+https://www.aliyun.com/product/ots
+
+## Maven Dependency
+
+```xml
+
+<dependency>
+    <groupId>dev.langchain4j</groupId>
+    <artifactId>langchain4j-tablestore</artifactId>
+    <version>{latest-version}</version>
```
Comment (langchain4j) [inline docs/docs/integrations/embedding-stores/tablestore.md]:
```suggestion
    <version>0.35.0</version>
```

File: langchain4j-tablestore/pom.xml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>dev.langchain4j</groupId>
+            <artifactId>langchain4j-core</artifactId>
+        </dependency>
+        
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
```
Comment (langchain4j) [inline langchain4j-tablestore/pom.xml]: please do not use lombok in new code

File: langchain4j-tablestore/src/main/java/dev/langchain4j/store/embedding/tablestore/TablestoreUtils.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package dev.langchain4j.store.embedding.tablestore;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import dev.langchain4j.internal.ValidationUtils;
+
+public class TablestoreUtils {
```
Comment (langchain4j) [inline langchain4j-tablestore/src/main/java/dev/langchain4j/store/embedding/tablestore/TablestoreUtils.java]:
```suggestion
class TablestoreUtils {
```

File: langchain4j-tablestore/src/main/java/dev/langchain4j/store/embedding/filter/comparison/IsTextMatch.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
+package dev.langchain4j.store.embedding.filter.comparison;
```
Comment (langchain4j) [inline langchain4j-tablestore/src/main/java/dev/langchain4j/store/embedding/filter/comparison/IsTextMatch.java]: please move it to `dev.langchain4j.store.embedding.tablestore` package

File: langchain4j-tablestore/src/main/java/dev/langchain4j/store/embedding/filter/comparison/IsTextMatchPhrase.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import dev.langchain4j.store.embedding.filter.Filter;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
+
+@ToString
+@EqualsAndHashCode
+public class IsTextMatchPhrase implements Filter {
```
Comment (langchain4j) [inline langchain4j-tablestore/src/main/java/dev/langchain4j/store/embedding/filter/comparison/IsTextMatchPhrase.java]: please move it to `dev.langchain4j.store.embedding.tablestore` package

**Review by ghost — COMMENTED** [review]
@xjtushilei thank you!

File: docs/docs/integrations/embedding-stores/tablestore.md (hunk 1 above)
Comment (xjtushilei) [reply docs/docs/integrations/embedding-stores/tablestore.md]: done.    I also upgraded the version in pom from 0.34.0-SNAPSHOT to 0.35.0-SNAPSHOT.

File: langchain4j-tablestore/pom.xml (hunk 2 above)
Comment (xjtushilei) [reply langchain4j-tablestore/pom.xml]: I removed all the lombok in this module

File: langchain4j-tablestore/src/main/java/dev/langchain4j/store/embedding/tablestore/TablestoreUtils.java (hunk 3 above)
Comment (xjtushilei) [reply langchain4j-tablestore/src/main/java/dev/langchain4j/store/embedding/tablestore/TablestoreUtils.java]: done

**Comment by xjtushilei** [comment]
@langchain4j  Thank you for your review. I have revised the above comments.

**Review by ghost — APPROVED** [review]
@xjtushilei thanks a lot! I hope you will maintain this module

**Pull request merged**